src/lib/types.ts

```diff
@@ -1197,3 +1282,275 @@ export type SignOut = {
+ * spec evolves.
+ *
+ * See https://www.w3.org/TR/webauthn-2/#iface-authenticatorattestationresponse
+ *
+ * Properties marked optional are not supported in all browsers.
+ */
+export interface AuthenticatorAttestationResponseFuture extends AuthenticatorAttestationResponse {
+  getTransports(): AuthenticatorTransportFuture[]
+}
+
+/**
+ * A super class of TypeScript's `AuthenticatorTransport` that includes support for the latest
+ * transports. Should eventually be replaced by TypeScript's when TypeScript gets updated to
+ * know about it (sometime after 4.6.3)
+ */
+export type AuthenticatorTransportFuture =
+  | 'ble'
+  | 'cable'
+  | 'hybrid'
+  | 'internal'
+  | 'nfc'
+  | 'smart-card'
+  | 'usb'
+
+/**
+ * A super class of TypeScript's `PublicKeyCredentialDescriptor` that knows about the latest
+ * transports. Should eventually be replaced by TypeScript's when TypeScript gets updated to
+ * know about it (sometime after 4.6.3)
+ */
+export interface PublicKeyCredentialDescriptorFuture
+  extends Omit<PublicKeyCredentialDescriptor, 'transports'> {
+  transports?: AuthenticatorTransportFuture[]
+}
+
+/** */
+export type PublicKeyCredentialJSON = RegistrationResponseJSON | AuthenticationResponseJSON
+
+/**
+ * A super class of TypeScript's `PublicKeyCredential` that knows about upcoming WebAuthn features
+ */
+export interface PublicKeyCredentialFuture extends PublicKeyCredential {
+  type: PublicKeyCredentialType
+  // See https://github.com/w3c/webauthn/issues/1745
+  isConditionalMediationAvailable?(): Promise<boolean>
+  // See https://w3c.github.io/webauthn/#sctn-parseCreationOptionsFromJSON
+  parseCreationOptionsFromJSON?(
+    options: PublicKeyCredentialCreationOptionsJSON
+  ): PublicKeyCredentialCreationOptions
+  // See https://w3c.github.io/webauthn/#sctn-parseRequestOptionsFromJSON
+  parseRequestOptionsFromJSON?(
+    options: PublicKeyCredentialRequestOptionsJSON
+  ): PublicKeyCredentialRequestOptions
+  // See https://w3c.github.io/webauthn/#dom-publickeycredential-tojson
+  toJSON?(): PublicKeyCredentialJSON
+}
+
+/**
+ * A custom Error used to return a more nuanced error detailing _why_ one of the eight documented
+ * errors in the spec was raised after calling `navigator.credentials.create()` or
+ * `navigator.credentials.get()`:
+ *
+ * - `AbortError`
+ * - `ConstraintError`
+ * - `InvalidStateError`
+ * - `NotAllowedError`
+ * - `NotSupportedError`
+ * - `SecurityError`
+ * - `TypeError`
+ * - `UnknownError`
+ *
+ * Error messages were determined through investigation of the spec to determine under which
+ * scenarios a given error would be raised.
+ */
+export class WebAuthnError extends Error {
+  code: WebAuthnErrorCode
+
+  constructor({
+    message,
+    code,
+    cause,
+    name,
+  }: {
+    message: string
+    code: WebAuthnErrorCode
+    cause: Error
+    name?: string
+  }) {
+    // @ts-ignore: help Rollup understand that `cause` is okay to set
```

Comment:
The build fails because of this ts-ignore comment in this helper function, which is ported over from simple-webauthn.